src/handlers/executors/vault.ts

```diff
@@ -109,21 +133,40 @@ export const handleVaultPhaseUpdated = async ({
   try {
     await vaultRecord.save()
 
-    logger.info(`#${header.number.toNumber()} handle VaultPhaseUpdated: ${JSON.stringify(vaultRecord)}`)
+    logger.info(
+      `#${header.number.toNumber()} handle VaultPhaseUpdated: ${JSON.stringify(
+        vaultRecord
+      )}`
+    )
   } catch (error) {
     logger.error('handle VaultPhaseUpdated error: ', error)
   }
 }
 
 export const handleVaultDissolved = async ({
+  idx,
   event: { data },
-  block: { block: { header }},
+  block: {
+    block: { header }
+  }
 }: SubstrateEvent) => {
   const [paraId, vaultId] = JSON.parse(data.toString()) as [number, number[]]
 
-  let vault = aggregateIntoId(paraId.toString(), vaultId[0].toString(), vaultId[1].toString());
+  let aggregateVaultId = aggregateIntoId(
+    paraId.toString(),
+    vaultId[0].toString(),
+    vaultId[1].toString()
+  )
   try {
-    await Vaults.remove(vault)
+    await VaultsDissolve.create({
+      id: idx.toString(),
+      vaultId: aggregateVaultId,
+      dissolvedBlockHeight: header.number.toNumber()
+    }).save()
+
+    let vault = await Vaults.get(aggregateVaultId)
+    vault.phase = 'Cancelled'
```

Comment:
Maybe change to `Dissolved`?

Reply:
It's better to make consist with the enum in pallet.